### PR TITLE
Plausible Analytics + Custom Events

### PR DESCRIPTION
**Pull request by codemonkey800**
## Description

Sets up Plausible analytics and sending custom events for the hub. This uses [next-plausible](https://github.com/4lejandrito/next-plausible/) to integrate plausible with our Next.js app.

## Changes

- Added environment variables
  - `ENV`: Denotes the deployment environment. Possible values are `local`, `dev`, `staging`, and `prod`. By default, its value is set to `local`
  - `PLAUSIBLE`: Enables or disables plausible analytics. Possible values are `true` and `false`. This can be used to force enable analytics for testing. By default, it's set to `true` only for prod and staging environments.
- `usePlausible()` hook: Wrapper hook that adds typing for custom events
- Added custom events for:
  - Search Page:
    - Searching
    - Filtering
    - Sorting
  - Plugin page:
    - Clicking on TOC sections
    - Clicking on outbound links
    - Clicking on the install button
    - Clicking on description nav'

## Demos

Demo Link: https://napari-hub-plausible-demo.vercel.app/
Plausible Dashboard: https://plausible.io/dev.napari-hub.org

**Note**: The plugin data is fixed for the demo, so every plugin page will show the data for `napari-compressed-io`

### Analytics

image

### Events

image

image

### CODE REVIEW

**Comment by neuromusic** [comment]
@codemonkey800 interesting. that's disappointing.

i had assumed that (minimally) it would work like the filters for OS and other metadata... I can select "Windows" under OS then click on "Browser" and see the breakdown for Browser use among windows users

image

I expected that I would be able to (say) select the "Filter" goal then the "3.7" and see the breakdown by True/False similarly.

**Comment by neuromusic** [comment]
nonetheless, the data appears to be there, so hopefully we can query it ourselves through the API to do any analysis that requires correlating across the custom properties for the goals.

**Comment by neuromusic** [comment]
https://github.com/plausible/analytics/discussions/846

**Comment by codemonkey800** [comment]
@neuromusic Sounds good, and provided we have the ability to query the analytics with all the original data, we can always build something out using the data. But it also sounds like they're aware of the issue so I'm hoping for more powerful features with filtering the event data.

**Review by kne42 — APPROVED** [review]
LGTM 👍

File: client/src/pages/_app.tsx [hunk 1]
```diff
@@ -45,19 +46,57 @@ function ReactQueryProvider({ children, dehydratedState }: QueryProviderProps) {
...
+}
+
+function PlausibleProvider({ children }: ProviderProps) {
+  const isUsingPlausible = process.env.PLAUSIBLE === 'true';
+  if (!isUsingPlausible) {
+    return <>{children}</>;
+  }
+
+  const isProd = process.env.ENV === 'prod';
+  const domain = isProd ? 'napari-hub.org' : 'dev.napari-hub.org';
```
Comment (justinelarsen) [inline client/src/pages/_app.tsx:85]: Can you add a comment on this? It looks like this doesn't actually matter what the domain is but it's rather a string that is an id in Plausible configuration setup?

**Review by justinelarsen — APPROVED** [review]
Nice work @codemonkey800 ! I just left some minor comments/questions, but looks good overall.

File: client/next.config.js [hunk 2]
```diff
@@ -27,6 +28,19 @@ module.exports = withMDX({
       config.resolve.alias.lodash = require.resolve('lodash-es');
     }
 
+    config.plugins.push(
+      new EnvironmentPlugin({
+        // Environment variable for current deployment environment (possible
+        // values are local, dev, staging, and prod)
+        ENV: 'local',
```
[the remark this answers is not in the document]
Comment (codemonkey800) [reply client/next.config.js:35]: That's correct! I'll add a comment to make it clear

**Pull request merged**